sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -624,7 +635,25 @@ public void onCallSucceeded(String payload, Map<String, String> headers) {
                                 .setCurrentPage(page).setTokenResult(tokenResult)
                                 .setHttpClient(mHttpClient)
                                 .setContinuationToken(headers.get(Constants.CONTINUATION_TOKEN_HEADER))
-                                .setDocumentType(documentType);
+                                .setReadOptions(readOptions)
+                                .setDocumentType(documentType)
+                                .setNextPageDelegate(new NextPageDelegate() {
+                                    
+                                    @Override
+                                    public <T> void loadNextPage(
```

Comment:
Need to rename T within this scope, there is a warning (just for loadNextPage here).